metals/src/main/scala/scala/meta/internal/metals/DocumentHighlightProvider.scala

```diff
@@ -0,0 +1,162 @@
+        else if (inf.isParameter && (inf.symbol.contains("apply") || inf.symbol
+            .contains("copy"))) {
+          parameterAlternatives(inf)
+        } else if (inf.isMethod) {
+          methodAlternatives(inf)
+        } else {
+          Set.empty
+        }
+      case None => Set.empty
+    }
+  }
+
+  private def methodAlternatives(info: SymbolInformation): Set[String] = {
+
+    def isInObject(desc: Descriptor) = desc match {
+      case Descriptor.Term(value) => true
+      case _ => false
+    }
+
+    val setterSuffix = "_="
+    Symbol(info.symbol) match {
+      case GlobalSymbol(
+          GlobalSymbol(owner, descriptor),
+          Descriptor.Method(setter, disambiguator)
+          ) =>
+        generateAlternativeSymbols(
+          setter.stripSuffix(setterSuffix),
+          descriptor.value,
+          owner.value,
+          isInObject(descriptor)
+        )
+      case GlobalSymbol(
+          GlobalSymbol(owner, descriptor),
+          Descriptor.Term(name)
+          ) =>
+        generateAlternativeSymbols(
+          name,
+          descriptor.value,
+          owner.value,
+          isInObject(descriptor)
+        )
+    }
+  }
+
+  private def parameterAlternatives(info: SymbolInformation): Set[String] = {
+    val copyOrApply = Set("apply", "copy")
+    Symbol(info.symbol) match {
+      case GlobalSymbol(
+          GlobalSymbol(
+            GlobalSymbol(owner, descriptor),
+            Descriptor.Method(name, disambiguator)
+          ),
+          desc
+          ) if copyOrApply(name) =>
+        generateAlternativeSymbols(
+          desc.value,
+          descriptor.value,
+          owner.value,
+          areParamsInObject = false
+        )
+
+      case _ =>
+        Set.empty
+    }
+  }
+
+  private def generateAlternativeSymbols(
+      paramName: String,
+      className: String,
+      packageName: String,
+      areParamsInObject: Boolean
+  ): Set[String] = {
+    val setterSuffix = "_="
+    val paramsDecriptor =
+      if (areParamsInObject) Descriptor.Term(className)
+      else Descriptor.Type(className)
+    Set(
+      Symbols.Global(
+        Symbols.Global(packageName, Descriptor.Type(className)),
+        Descriptor.Term(paramName)
+      ),
+      Symbols.Global(
+        Symbols.Global(
+          Symbols.Global(packageName, Descriptor.Type(className)),
+          Descriptor.Method("copy", "()")
+        ),
+        Descriptor.Parameter(paramName)
+      ),
+      Symbols.Global(
+        Symbols.Global(
+          Symbols.Global(packageName, Descriptor.Term(className)),
+          Descriptor.Method("apply", "()")
+        ),
+        Descriptor.Parameter(paramName)
+      ),
+      Symbols.Global(
+        Symbols.Global(packageName, paramsDecriptor),
+        Descriptor.Method(paramName, "()")
+      ),
+      Symbols.Global(
+        Symbols.Global(packageName, paramsDecriptor),
+        Descriptor.Method(paramName + setterSuffix, "()")
+      )
+    )
+  }
```

Comment:
Can we get `MatchError` here?

Reply:
this method is just used to generate all possible alternatives - no matches here.

Reply:
Whoops, look good!